src/components/NcAppContent/NcAppContent.vue

```diff
@@ -268,6 +284,24 @@ export default {
 				},
 			}
 		},
+
+		realPageTitle() {
+			if (this.pageTitle) {
+				return this.pageTitle
+			}
+			if (this.pageHeading) {
+				return t('{view} - {app} - {product}', { view: this.pageHeading, app: localizedAppName, product: productName })
+			}
+			return null
+		},
+	},
+
+	watch: {
+		realPageTitle() {
+			if (this.realPageTitle !== null) {
+				window.document.title = this.realPageTitle
+			}
+		},
```

Comment:
Shouldn't it be `immediate`? Otherwise, it only works on changing title/heading but not setting.

Or do I miss something?

Reply:
No you are right, fixed!